### PR TITLE
Closes #22 - Fix permission attributes;

### LINKED ISSUE #22: Fix the error that will occur if permission attributes are empty

### PR DESCRIPTION
**Pull request by ryzheboka**
no error should occur in case they are empty

### Thanks for your PR! Please fill out the following list :)

---

- [x] I put the ticket or multiple tickets in review
- [x] Commit message format → Closes #&lt;Issue Number&gt; - Your commit message.
- [ ] Sonarcloud link :  https://sonarcloud.io/summary/new_code?id=ryzheboka_kadai&branch=KAD-22b
- [x] No documentation update needed
- [ ] Link to PR with documentation update: \<add the link here>
- [x] No Release Notes needed
- [ ] Release Notes :

```

```

### CODE REVIEW

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java [hunk 1]
```diff
@@ -907,6 +913,22 @@ private Set<String> getPermissionIdsFromContext(final DirContextOperations conte
     return permissions;
   }
 
+  private boolean permissionsAreEmpty() {
+    return getUserPermissionsAttribute().isEmpty() || getPermissionSearchFilterName().isEmpty();
```
Comment (CRoberto1926) [inline rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: I guess we can get null pointers here, or am I wrong?

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java [hunk 2]
```diff
@@ -907,6 +913,22 @@ private Set<String> getPermissionIdsFromContext(final DirContextOperations conte
...
+  }
+
+  private AndFilter getPermissionsNotPresentAndFilter(AndFilter andFilter) {
+    final AndFilter andFilter2;
+    if (!getUserPermissionsAttribute().isEmpty()) {
```
Comment (CRoberto1926) [inline rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: Null pointer?

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java [hunk 3]
```diff
@@ -985,7 +1007,8 @@ public AccessIdRepresentationModel doMapFromContext(final DirContextOperations c
         String firstName = context.getStringAttribute(getUserFirstnameAttribute());
         String lastName = context.getStringAttribute(getUserLastnameAttribute());
         accessId.setName(String.format("%s, %s", lastName, firstName));
-      } else if (context.getStringAttribute(getUserPermissionsAttribute()) == null) {
+      } else if (getUserPermissionsAttribute().isEmpty()
```
Comment (CRoberto1926) [inline rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: Null pointer?

**Review by holgerhagen — CHANGES_REQUESTED** [review]
I would like to have a test verifying that the filter conditions is built correctly. Or doesn't this make sense here?

**Comment by ryzheboka** [comment]
> I would like to have a test verifying that the filter conditions is built correctly. Or doesn't this make sense here?

This is party verified by different LDAP tests by testing the general methods like searchUsersAndGroupsAndPermissions that use the filter. Testing just the filter separately doesn't make sense to me. However, the LDAP tests depend on the configured values for the LDAP attributes like UserMemberOfGroupAttribute etc.

I could change the values of the relevant attributes to be empty for one of the tests. Would this help?

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java (hunk 1 above)
Comment (ryzheboka) [reply rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: I am not sure. I will add a null check to be sure

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java (hunk 2 above)
Comment (ryzheboka) [reply rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: I will add a check

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java (hunk 3 above)
Comment (ryzheboka) [reply rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: I will add a check

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java [hunk 4]
```diff
@@ -907,6 +913,23 @@ private Set<String> getPermissionIdsFromContext(final DirContextOperations conte
...
+    final AndFilter andFilter2;
+    if (getUserPermissionsAttribute() != null && !getUserPermissionsAttribute().isEmpty()) {
+      andFilter2 = new AndFilter();
+      andFilter2.and(new NotPresentFilter(getUserPermissionsAttribute()));
+      andFilter2.and(andFilter);
+    } else {
+      andFilter2 = andFilter;
+    }
+    return andFilter2;
+  }
```
Comment (CRoberto1926) [inline rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]:
```suggestion
  private AndFilter getPermissionsNotPresentAndFilter(AndFilter andFilter) {

    if (getUserPermissionsAttribute() == null || getUserPermissionsAttribute().isEmpty()) {
      return andFilter;
    }

    final AndFilter
        andFilter2 = new AndFilter();
    andFilter2.and(new NotPresentFilter(getUserPermissionsAttribute()));
    andFilter2.and(andFilter);
    return andFilter2;
  }
```

**Review by CRoberto1926 — APPROVED** [review]

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java (hunk 4 above)
Comment (ryzheboka) [reply rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: That's a good idea!

**Review by holgerhagen — APPROVED** [review]

**Pull request merged**